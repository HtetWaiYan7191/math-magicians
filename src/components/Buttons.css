.input-button {
  border: 1px solid rgb(15, 23, 42);
  background-color: rgb(30 41 59);
  padding: 10px 12px;
  color: #22d3ee;
  font-size: 20px;
  text-align: center;
}

.input-button:hover {
  background-color: #22d3ee;
  color: black;
}

.zero {
  grid-column: 1/3;
}

.orange {
  background-color: rgb(15, 23, 42);
  color: #22d3ee;
  font-size: 1.5rem;
  border: 1px solid #22d3ee;
}

.orange:hover {
  border: 1px solid black;
  background-color: #22d3ee;
  color: black;
}
