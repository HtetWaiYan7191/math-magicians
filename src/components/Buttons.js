import React from 'react';
import './Buttons.css';
import PropTypes from 'prop-types';

const Buttons = ({ number }) => {
  if (number === '\u00F7' || number === 'x' || number === '-' || number === '+' || number === '=') {
    return <button type="button" className="input-button orange">{number}</button>;
  } if (number === '0') {
    return <button type="button" className="input-button zero">{number}</button>;
  }
  return <button type="button" className="input-button">{number}</button>;
};

Buttons.propTypes = {
  number: PropTypes.number.isRequired,
};

export default Buttons;
