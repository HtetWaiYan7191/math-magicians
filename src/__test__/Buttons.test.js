import React from 'react';
import renderer from 'react-test-renderer';
import Buttons from '../components/Buttons';

describe('Buttons', () => {
  test('should render the Button component', () => {
    const handleInput = jest.fn();

    const component = renderer.create(<Buttons number="1" handleInput={handleInput} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    tree = component.root;
    expect(tree.findByType(Buttons).props.number).toBe('1');
  });
});
