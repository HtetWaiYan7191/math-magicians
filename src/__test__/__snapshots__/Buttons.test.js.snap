// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons should render the Button component 1`] = `
<button
  className="input-button"
  onClick={[MockFunction]}
  type="button"
>
  1
</button>
`;
